packages/edit-post/src/components/layout/style.scss

```
@@ -1,6 +1,20 @@
 .edit-post-layout__metaboxes {
 	flex-shrink: 0;
 	clear: both;
+	overflow: auto;
+	filter: drop-shadow(0 -1px rgba($color: #000, $alpha: 0.133)); // 0.133 = $gray-200 but with alpha.;
+
+	&:has(.postbox:not(.closed)) {
+		flex: 1 1 50%;
+	}
```

```suggestion
	&:has(.postbox:not(.closed):not(.is-hidden)) {
		flex: 1 1 50%;
	}
```

If a meta box is expanded and then disabled in the preferences panel, the metabox content will be visually hidden but it doesn't have the `.closed` class, so an empty meta box will be rendered. To prevent this, we might also want to check for the absence of the `.is-hidden` class:

https://github.com/user-attachments/assets/74b3b895-e788-456f-8ca0-45c063c56485

---

Spot on! 🙌